### PR TITLE
Added new route for obtaining docInfo from fileId

### PR DESCRIPTION
**Pull request by merttumer**
Signed-off-by: Mert Tumer <mert.tumer@collabora.com>

### CODE REVIEW

**Comment by merttumer** [comment]
@pmaier1 @kulmann Can you please review this, somehow I cant add you as reviewers nor I can merge it, Thank you

File: js/web/app.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+              return $gettext("Document")
+            },
+            icon: "x-office-document"
+          },
+          routes: [
+            "files-personal",
+            "files-favorites",
+            "files-shared-with-others",
+            "files-shared-with-me",
+            "files-public-list"
```
Comment (kulmann) [inline js/web/app.js:35]: Your current implementation doesn't support public links - which is fine for a first iteration. But please remove `files-public-list` from the routes whitelist, so that collabora is not offered as editor on public links for now. Same for the other file extensions.

File: js/web/editor.vue [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+                headers: {
+                    Authorization: "Bearer " + this.getToken,
+                },
+            }).then((response) => {
+                return response.data;
+            });
+        },
+
+        generateWOPISrc(documentInfo) {
+            var documentId =
```
Comment (kulmann) [inline js/web/editor.vue:112]: this can be simplified to 
```js
const documentId = [documentInfo.fileId, documentInfo.instanceId, documentInfo.version, documentInfo.sessionId].join('_')
```

File: js/web/editor.vue [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+                },
+            });
+        },
+
+        cancelSaveAs() {
+            this.hideSaveAs = true;
+        },
+
+        confirmSaveAs(value) {
+            if (value) {
```
Comment (JanAckermann) [inline js/web/editor.vue:64]: Early return :) ? What should happen g
If value is not set or false ?

File: js/web/editor.vue [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+        // where we get document url from owncloud api
+        async getDocumentFileInfo() {
+            return axios({
+                method: "GET",
+                url:
+                    this.configuration.server +
+                    "index.php/apps/richdocuments/ajax/documents/index/" +
+                    this.fileId,
+                headers: {
+                    Authorization: "Bearer " + this.getToken,
```
Comment (JanAckermann) [inline js/web/editor.vue:104]: Wondering why getToken is not a function call,? If it is a computed,  getToken seems not right name.

File: js/web/editor.vue [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    },
+
+    methods: {
+        ...mapActions(["showMessage"]),
+
+        messageDisplay(desc, status = "danger", title = "") {
+            this.showMessage({
+                title: title,
+                desc: desc,
+                status: status,
```
Comment (JanAckermann) [inline js/web/editor.vue:52]:
```suggestion
                status,
```

File: js/web/editor.vue [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+            return axios({
+                method: "GET",
+                url:
+                    this.configuration.server +
+                    "index.php/apps/richdocuments/ajax/documents/index/" +
+                    this.fileId,
+                headers: {
+                    Authorization: "Bearer " + this.getToken,
+                },
+            }).then((response) => {
```
Comment (JanAckermann) [inline js/web/editor.vue:106]: Could be extracted to {data} in arrrow func param

File: js/web/app.js [hunk 7]
```diff
@@ -28,23 +28,15 @@ const appInfo = {
             icon: "x-office-document"
           },
           routes: [
-            "files-personal",
-            "files-favorites",
-            "files-shared-with-others",
-            "files-shared-with-me",
-            "files-public-list"
+            "files-personal"
```
Comment (kulmann) [inline js/web/app.js:31]: Why did you remove all the route checks? The `routes` array defines the routes on which Collabora will be available. Should be working for `files-favorites`, `files-shared-with-others` and `files-shared-with-me` (at least for accepted shares). The public links page needs testing, not sure if it works there. If not you'd be right to remove it from the array of allowed routes.

**Comment by phil-davis** [comment]
@merttumer the current oc10 core master and the https://github.com/owncloud/richdocuments/commits/master have dropped PHP 7.2 in recent weeks.

Your branch in the fork is out-of-date, and is still trying to run some PHP 7.2 pipelines in CI. And those fail like:
https://drone.owncloud.com/owncloud/richdocuments/1650/5/3
because the core oC10 fetched by the CI now requires  PHP  7.3 or 7.4.

You need to rebase your fork branch. But that will be more difficult because your commits are directly in the `master` branch in your fork.

Are you able to sort out getting your commits into a separate branch in your fork, and getting `master` up-to-date with https://github.com/owncloud/richdocuments ?

**Comment by merttumer** [comment]
> @merttumer the current oc10 core master and the https://github.com/owncloud/richdocuments/commits/master have dropped PHP 7.2 in recent weeks.
> 
> Your branch in the fork is out-of-date, and is still trying to run some PHP 7.2 pipelines in CI. And those fail like: https://drone.owncloud.com/owncloud/richdocuments/1650/5/3 because the core oC10 fetched by the CI now requires PHP 7.3 or 7.4.
> 
> You need to rebase your fork branch. But that will be more difficult because your commits are directly in the `master` branch in your fork.
> 
> Are you able to sort out getting your commits into a separate branch in your fork, and getting `master` up-to-date with https://github.com/owncloud/richdocuments ?

Ah, np. I can handle it. Thank you

**Comment by phil-davis** [comment]
https://github.com/merttumer/richdocuments-1
"This branch is 9 commits ahead of owncloud:master. "
Great stuff! And CI passes.

@kulmann @JanAckermann @kendy please review.

**Comment by mmeeks** [comment]
Would love to get this in & released finally ! =) any more feedback ?

**Review by JammingBen — CHANGES_REQUESTED** [review]
Few small things left from my side, rest looks good!

File: lib/Controller/DocumentController.php [hunk 8]
```diff
@@ -469,6 +469,23 @@ private function handleDocIndex($fileId, $shareToken, $currentUser) {
...
+	/**
+	 * Accessed from external-apps such as new owncloud web front-end
+	 * It returns the information to load the document from the fileId.
+	 * @NoAdminRequired
+	 * @CORS
+	 * @NoCSRFRequired
+	 */
+	public function getDocumentIndex($fileId) {
+		try {
+			$docRetVal = $this->handleDocIndex($fileId, null, \OC_User::getUser());
```
[the remark this answers is not in the document]
Comment (merttumer) [reply lib/Controller/DocumentController.php:481]: well, $this->uid was my first choice, but unfortunately, there's a problem with it and it becomes null on CORS requests. Only \OC_User::getUser() works. Otherwise we get bad request.

File: lib/Controller/DocumentController.php [hunk 9]
```diff
@@ -469,6 +469,23 @@ private function handleDocIndex($fileId, $shareToken, $currentUser) {
...
+			$docRetVal["locale"] = \strtolower(\str_replace('_', '-', $this->settings->getUserValue(\OC_User::getUser(), 'core', 'lang', 'en')));
```
[the remark this answers is not in the document]
Comment (merttumer) [reply lib/Controller/DocumentController.php:482]: well, $this->uid was my first choice, but unfortunately, there's a problem with it and it becomes null on CORS requests. Only \OC_User::getUser() works. Otherwise we get bad request.

**Comment by merttumer** [comment]
any more feedback, please? @JammingBen @kulmann @pmaier1

File: lib/Controller/WebAssetController.php [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+	private $logger;
+
+	/**
+	 * WebAssetController constructor.
+	 *
+	 * @param string $AppName - application name
+	 * @param IRequest $request - request object
+	 * @param ILogger $logger
+	 */
+	public function __construct($AppName, IRequest $request, ILogger $logger) {
```
Comment (JanAckermann) [inline lib/Controller/WebAssetController.php:47]: $AppName might be lowerCamelCase

**Review by JammingBen — APPROVED** [review]

**Review by pmaier1 — COMMENTED** [review]
Thanks! Please merge and we'll start releasing richdocuments.

**Review by kulmann — APPROVED** [review]
still one more small finding - approving because it's not critical.

**Review by JanAckermann — APPROVED** [review]
